js/plugins/iq.js

```diff
@@ -0,0 +1,352 @@
+  
+  this.ooyalaReporter = null;
+  this.testMode = false;
+
+  /**
+   * [Required Function] Return the name of the plugin.
+   * @public
+   * @method IqPlugin#getName
+   * @return {string} The name of the plugin.
+   */
+  this.getName = function ()
+  {
+    return name;
+  };
+
+  /**
+   * [Required Function] Return the version string of the plugin.
+   * @public
+   * @method IqPlugin#getVersion
+   * @return {string} The version of the plugin.
+   */
+  this.getVersion = function ()
+  {
+    return version;
+  };
+
+  /**
+   * Return the autoPlay value.
+   * @public
+   * @method IqPlugin@getAutoPlay
+   * @return {boolean} The value of autoPlay.
+   */
+  this.getAutoPlay = function()
+  {
+    return autoPlay;
+  };
+
+  /**
+   * [Required Function] Set the plugin id given by the Analytics Framework when
+   * this plugin is registered.
+   * @public
+   * @method IqPlugin#setPluginID
+   * @param  {string} newID The plugin id
+   */
+  this.setPluginID = function(newID)
+  {
+    id = newID;
+  };
+
+  /**
+   * [Required Function] Returns the stored plugin id, given by the Analytics Framework.
+   * @public
+   * @method IqPlugin#setPluginID
+   * @return  {string} The pluginID assigned to this instance from the Analytics Framework.
+   */
+  this.getPluginID = function()
+  {
+    return id;
+  };
+
+  /**
+   * [Required Function] Initialize the plugin with the given metadata.
+   * @public
+   * @method IqPlugin#init
+   */
+  this.init = function()
+  {
+    var missedEvents;
+    //if you need to process missed events, here is an example
+    if (_framework && OO._.isFunction(_framework.getRecordedEvents))
+    {
+      missedEvents = _framework.getRecordedEvents();
+    }
+    //use recorded events.
+
+    if (this.testMode)
+    {
+      trySetupAnalytics();
+    }
+    else if (!window.Ooyala)
+    {
+      OO.loadScriptOnce("https://analytics.ooyala.com/static/v3/analytics.js", trySetupAnalytics, sdkLoadError, SDK_LOAD_TIMEOUT);
+    }
+  };
+
+  /**
+   * [Required Function] Set the metadata for this plugin.
+   * @public
+   * @method IqPlugin#setMetadata
+   * @param  {object} metadata The metadata for this plugin
+   */
+  this.setMetadata = function(metadata)
+  {
+      OO.log( "Analytics Template: PluginID \'" + id + "\' received this metadata:", metadata);
+  };
+
+  /**
+   * [Required Function] Process an event from the Analytics Framework, with the given parameters.
+   * @public
+   * @method IqPlugin#processEvent
+   * @param  {string} eventName Name of the event
+   * @param  {Array} params     Array of parameters sent with the event
+   */
+  this.processEvent = function(eventName, params)
+  {
+    OO.log( "Analytics Template: PluginID \'" + id + "\' received this event \'" + eventName + "\' with these params:", params);
+    switch(eventName)
+    {
+      //OO.EVENTS.AUTHORIZATION_FETCHED -> OO.Analytics.STREAM_TYPE_UPDATED
+      case OO.Analytics.EVENTS.STREAM_TYPE_UPDATED:
```

Comment:
Are we doing anything with this event? We can remove this if it is not needed.

Reply:
Nope. This was in the original code for the plugin. Should I remove it completely? We could also just comment it out for now.

Reply:
I just checked for other instances of STREAM_TYPE_UPDATED, it looks like this is actually part of the analytics plugin template

Reply:
We should remove this if it is not being used. I think the template has it to showcase how to use the stream type.

Reply:
Removed